Writerside/topics/analysis/domain/raptor.md

```diff
@@ -29,49 +29,34 @@ graphs that incorporate the complexity of transfers. This characteristic makes R
 networks of any size, as it avoids the need for time-consuming preprocessing. This efficiency allows us to enable
 features like accessibility and bike information on our routes.
 
-## Sequence Diagram RAPTOR (simple)
+## Sequence Diagram RAPTOR (simplified)
 
 ```mermaid
 sequenceDiagram
-    participant User
-    participant RAPTOR Algorithm
-    participant Routes
-    User ->> RAPTOR Algorithm: Request journey plan
-    RAPTOR Algorithm ->> Routes: Initialize routes and transfers
-    Routes -->> RAPTOR Algorithm: Routes initialized
-
-    loop For each round k
-        RAPTOR Algorithm ->> Routes: Process marked stops and routes
-        Routes -->> RAPTOR Algorithm: Return earliest trip information
-        RAPTOR Algorithm ->> RAPTOR Algorithm: Update arrival times
+  participant Algorithm
```

Comment:
@clukas1 

simplified, conceptional sequence diagram, based on pseudocode


```mermaid
sequenceDiagram
  participant Algorithm
  participant Stops
  participant Routes
  participant Trips
  participant FootPaths
  Algorithm ->> Stops: Initialize and mark start stop

  loop Main iterations (k)
    Algorithm ->> Stops: Identify marked stops
    Stops ->> Routes: Get routes serving marked stops
    Routes ->> Trips: Traverse routes and trips

    alt Can update stops?
      Trips ->> Stops: Update stop times and mark new stops
    end

    Algorithm ->> FootPaths: Check foot-path connections
    FootPaths ->> Stops: Update stops via foot-paths

    alt No stops updated
      Algorithm ->> Algorithm: Stop process
    end
  end

```

Reply:
I think this is better, couple of minor suggestions:

- Add initial footpath relaxation after "initialize and mark start stop"
- Check foot-path connections is also done from Stops if you say Get Routes serving marked stops originates from Stops.
- Same as for route scanning, footpath relaxations should also only update marked stops when improvements can be achieved (alt condition)